packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -214,41 +303,62 @@ export class Fleet extends Resource implements IFleet {
       throw new ValidationError('baseCapacity must be greater than or equal to 1', this);
     }
 
+    let computeConfiguration: CfnFleet.ComputeConfigurationProperty | undefined;
     if (
-      props.computeType === FleetComputeType.ATTRIBUTE_BASED &&
-      (!props.computeConfiguration || Object.keys(props.computeConfiguration).length === 0)
+      props.computeType === FleetComputeType.ATTRIBUTE_BASED
     ) {
-      throw new ValidationError('At least one compute configuration criteria must be specified if computeType is "ATTRIBUTE_BASED"', this);
-    }
-    if (props.computeConfiguration && props.computeType !== FleetComputeType.ATTRIBUTE_BASED) {
-      throw new ValidationError(`'computeConfiguration' can only be specified if 'computeType' is 'ATTRIBUTE_BASED', got: ${props.computeType}`, this);
+      if (!props.computeConfiguration || !(props.computeConfiguration.disk || props.computeConfiguration.machineType || props.computeConfiguration.memory || 'vCpu' in props.computeConfiguration)) {
+        throw new ValidationError('At least one compute configuration criteria must be specified if computeType is ATTRIBUTE_BASED', this);
+      } else if ('instanceType' in props.computeConfiguration) {
```

Comment:
I think it is better to implement like below.

```suggestion
      } else if (props.computeConfiguration.instanceType) {
```